java/client/src/main/java/glide/utils/ArrayTransformUtils.java

```diff
@@ -106,14 +106,31 @@ public static <T, U extends T> U[][] castArrayofArrays(T[] outerObjectArr, Class
      * @param <T> The base type from which the elements are being cast.
      * @param <U> The subtype of T to which the elements are cast.
      */
-    @SuppressWarnings("unchecked")
     public static <T, U extends T> Map<String, U[]> castMapOfArrays(
             Map<String, T[]> mapOfArrays, Class<U> clazz) {
         if (mapOfArrays == null) {
             return null;
         }
         return mapOfArrays.entrySet().stream()
-                .collect(Collectors.toMap(k -> k.getKey(), e -> castArray(e.getValue(), clazz)));
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> castArray(e.getValue(), clazz)));
+    }
+
+    /**
+     * Maps a Map of Object[][] with value type T[][] to value of U[][].
+     *
+     * @param mapOfArrays Map of 2D Array values to cast.
+     * @param clazz The class of the array values to cast to.
+     * @return A Map of arrays of type U[][], containing the key/values from the input Map.
+     * @param <T> The base type from which the elements are being cast.
+     * @param <U> The subtype of T to which the elements are cast.
+     */
+    public static <T, U extends T> Map<String, U[][]> castMapOf2DArray(
```

Comment:
You can remove `T`, because it is always `Object`
```suggestion
    public static <U> Map<String, U[][]> castMapOf2DArray(
```
(and then rename `U` to `T`)
Same is true for `castMapOfArrays`

Reply:
Ok.  That's fine.  I'll change this too.